geronimo-json_1.0_spec/src/main/java/javax/json/spi/JsonProvider.java

```diff
@@ -28,6 +28,9 @@
 import javax.json.stream.JsonGeneratorFactory;
 import javax.json.stream.JsonParser;
 import javax.json.stream.JsonParserFactory;
+
+import org.apache.geronimo.osgi.locator.ProviderLocator;
```

Comment:
although i've solved this successful in other osgi bundles using maven-shade-plugin it does not work very well in this case - because one of the located classes is used as Bundle-Activator, and relocating the class breaks the magic of the bundle plugin generating the correct imports. after some puzzling i got it working: https://github.com/stefanseifert/geronimo-specs/blob/feature/GERONIMO-6560-json10-osgi-shade/geronimo-json_1.0_spec/pom.xml
but the solution is quite hacky and fragile - i advice against using it.

i would propose that we either use the original PR without relocating the classes (as in jaxrs) - or completely drop the support for geronimo osgi-support (which is older than the release of the osgi r5 spec, which is already old), ans switch for this spec jar and johnzon to an osgi-compatible way with capabilities. i've never tried it out, but in my understanding no special embedded classes are required in this case. aries spyfly has to be used then.